Failure of a test `semlock_test_failure` in MacOS M1

I get a failure for the `semlock_test_failure`:

```pytb
____________________________________________________________ test_semlock_failure _____________________________________________________________
[gw1] darwin -- Python 3.8.12 /Users/glemaitre/mambaforge/envs/dev/bin/python

    @pytest.mark.skipif(sys.platform == "win32", reason="UNIX test")
    def test_semlock_failure():
        from loky.backend.synchronize import SemLock, sem_unlink
    
        name = "loky-test-semlock"
>       sl = SemLock(0, 1, 1, name=name)

tests/test_synchronize.py:24: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <loky.backend.synchronize.SemLock object at 0x1079ce9d0>, kind = 0, value = 1, maxvalue = 1, name = 'loky-test-semlock'

    def __init__(self, kind, value, maxvalue, name=None):
        # unlink_now is only used on win32 or when we are using fork.
        unlink_now = False
        if name is None:
            # Try to find an unused name for the SemLock instance.
            for _ in range(100):
                try:
                    self._semlock = _SemLock(
                        kind, value, maxvalue, SemLock._make_name(), unlink_now
                    )
                except FileExistsError:  # pragma: no cover
                    pass
                else:
                    break
            else:  # pragma: no cover
                raise FileExistsError('cannot find name for semaphore')
        else:
>           self._semlock = _SemLock(
                kind, value, maxvalue, name, unlink_now
            )
E           FileExistsError: [Errno 17] File exists

loky/backend/synchronize.py:73: FileExistsError
```

--- a/tests/test_synchronize.py
+++ b/tests/test_synchronize.py
@@ -21,11 +21,15 @@ def test_semlock_failure():
     from loky.backend.synchronize import SemLock, sem_unlink
 
     name = "loky-test-semlock"
-    sl = SemLock(0, 1, 1, name=name)
+    try:
+        sl = SemLock(0, 1, 1, name=name)
 
-    with pytest.raises(FileExistsError):
-        SemLock(0, 1, 1, name=name)
-    sem_unlink(sl.name)
+        with pytest.raises(FileExistsError):
+            SemLock(0, 1, 1, name=name)
+    finally:
+        # Always clean-up the test semaphore to make this test independent of
+        # previous runs (successful or not).
+        sem_unlink(sl.name)
 
     with pytest.raises(FileNotFoundError):
         sl._semlock._rebuild(0, 0, 0, name)

--- a/tests/test_synchronize.py
+++ b/tests/test_synchronize.py
@@ -23,13 +23,14 @@ def test_semlock_failure():
     name = "loky-test-semlock"
     try:
         sl = SemLock(0, 1, 1, name=name)
+        assert sl.name == name
 
         with pytest.raises(FileExistsError):
             SemLock(0, 1, 1, name=name)
     finally:
         # Always clean-up the test semaphore to make this test independent of
         # previous runs (successful or not).
-        sem_unlink(sl.name)
+        sem_unlink(name)
 
     with pytest.raises(FileNotFoundError):
         sl._semlock._rebuild(0, 0, 0, name)

--- a/tests/test_synchronize.py
+++ b/tests/test_synchronize.py
@@ -22,6 +22,10 @@ def test_semlock_failure():
 
     name = "loky-test-semlock"
     try:
+        sem_unlink(name)
+    except FileNotFoundError:
+        pass
+    try:
         sl = SemLock(0, 1, 1, name=name)
         assert sl.name == name
 

